### PR TITLE
add pytorch 1.5 supported

### PR DESCRIPTION
**Pull request by lbin**

### CODE REVIEW

File: src/cuda/dcn_v2_cuda.cu [hunk 1]
```diff
@@ -104,7 +104,7 @@ dcn_v2_cuda_forward(const at::Tensor &input,
     const int block = 128;
     const int grid = (batch + block - 1) / block;
 
-    createBatchGemmBuffer<<<grid, block, 0, THCState_getCurrentStream(state)>>>(
+    createBatchGemmBuffer<<<grid, block, 0, c10::cuda::getCurrentCUDAStream()>>>(
```
Comment (IgorDavidyuk) [inline src/cuda/dcn_v2_cuda.cu:107]: Hi!
Could you please explain why the way getting a stream is changed? A link may be.
Thanks in advance.

Comment (lbin) [reply src/cuda/dcn_v2_cuda.cu:107]: https://github.com/pytorch/pytorch/commit/a5f01846c2bae0d78a5c0cd74ced20a07b8ef961#diff-1dcab1425fcb102bc8786f370fa99028

Comment (IgorDavidyuk) [reply src/cuda/dcn_v2_cuda.cu:107]: Also I tried to use you branch with torch1.5 and https://github.com/xingyizhou/CenterTrack
CenterTrack couldn't import DCN.
Then I switched to torch1.4 and it worked just fine

Comment (lbin) [reply src/cuda/dcn_v2_cuda.cu:107]: I tested this version on my other projects and also centertrack, my suggestion is remove all your codes and find & rm pkg you installed in your python/pip/conda env

**Comment by austinmw** [comment]
@lbin Hi, thanks for this PR. I tested it with 1.5.1 and it worked for me. However I tried to use it with 1.6.0 (nightly 1.6.0.dev20200623+cu101') and was unable to compile and got `RuntimeError: Error compiling objects for extension`.

I think several deprecation warnings in 1.5 turned into errors, like:

> warning: \u2018at::DeprecatedTypeProperties& at::Tensor::type() const\u2019 is deprecated: Tensor.type() is deprecated. Instead use Tensor.options(), which in many cases (e.g. in a constructor) is a drop-in replacement. If you were using data from type(), that is now available from Tensor itself, so instead of tensor.type().scalar_type(), use tensor.scalar_type() instead and instead of tensor.type().backend() use tensor.device(). [-Wdeprecated-declarations]

I'm a C++ novice myself, so if you happen to get any time to update this PR for future compatibility I'd really appreciate it.

(By the way using 1.6.0 for torch tracing support of dict output types)

**Review by MARMOTatZJU — COMMENTED** [review]
This worked for me on image buildt by [this dockerfile](https://github.com/MARMOTatZJU/deepo/blob/marmot/docker-dev/marmot-sot/Dockerfile.pytorch151-py36-cu102-cudnn7-marmot-sot), i.e., CUDA10.2 & CUDNN7 & PyTorch1.5.1
Thus consider merging it.

File: src/cuda/dcn_v2_cuda.cu (hunk 1 above)
Comment (shenyingying) [reply src/cuda/dcn_v2_cuda.cu:107]: > [pytorch/pytorch@a5f0184#diff-1dcab1425fcb102bc8786f370fa99028](https://github.com/pytorch/pytorch/commit/a5f01846c2bae0d78a5c0cd74ced20a07b8ef961#diff-1dcab1425fcb102bc8786f370fa99028)

我尝试了同样的方法不好用

**Comment by lbin** [comment]
Pls follow https://github.com/lbin/DCNv2/tree/pytorch_1.6 for 1.6 support

File: src/cuda/dcn_v2_cuda.cu (hunk 1 above)
Comment (shenyingying) [reply src/cuda/dcn_v2_cuda.cu:107]: > Hi!
> Could you please explain why the way getting a stream is changed? A link may be.
> Thanks in advance.
you success,can you share something this!

Comment (zhangyingbit) [reply src/cuda/dcn_v2_cuda.cu:107]: It works for my case, cuda: 10.2, pytorch: 1.5

**Pull request closed without merging**